src/emeus-constraint-layout.c

```diff
@@ -1908,9 +1917,8 @@ emeus_constraint_layout_child_get_preferred_size (EmeusConstraintLayoutChild *se
   int child_nat = 0;
   Variable *attr = NULL;
 
-  switch (orientation)
+  if (self->solver && orientation == GTK_ORIENTATION_HORIZONTAL)
```

Comment:
They should be set to 0 in GTK+ itself, AFAIR, so that people implementing the virtual functions don't have to explicitly do so.